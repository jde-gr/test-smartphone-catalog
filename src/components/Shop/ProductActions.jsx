import { Formik, Field, Form } from 'formik';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';

import { Actions } from './styles/ProductDetails.styles';
import { postItemToCart } from '../../store/cart-thunks';
import { Button } from './styles/ProductActions.styles';

const ProductActions = ({ id, options }) => {
  const dispatch = useDispatch();

  return (
    <Actions>
      <Formik
        initialValues={{
          storageInput: String(options.storages[0].code),
          colorInput: String(options.colors[0].code),
        }}
        onSubmit={async (values) => {
          await new Promise((r) => setTimeout(r, 500));
          dispatch(postItemToCart(id, values.colorInput, values.storageInput));
        }}
      >
        {({ values }) => (
          <Form>
            <div id='storageGroup'>Almacenamiento: </div>
            <div role='group' aria-labelledby='storageGroup'>
              {options.storages.map((storage) => (
                <label key={storage.code}>
                  <Field
                    type='radio'
                    name='storageInput'
                    value={String(storage.code)}
                  />
                  {storage.name}{' '}
                </label>
              ))}
            </div>

            <div id='colorGroup'>Color: </div>
            <div role='group' aria-labelledby='colorGroup'>
              {options.colors.map((color) => (
                <label key={color.code}>
                  <Field
                    type='radio'
                    name='colorInput'
                    value={String(color.code)}
                  />
                  {color.name}{' '}
                </label>
              ))}
            </div>

            <Button type='submit'>AÃ±adir al carrito</Button>
          </Form>
        )}
      </Formik>
    </Actions>
  );
};

ProductActions.propTypes = {
  id: PropTypes.string,
  options: PropTypes.object,
};

export default ProductActions;
