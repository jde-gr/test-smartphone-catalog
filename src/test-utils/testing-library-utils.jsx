import { render } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../store';

const queryClient = new QueryClient();

const AllTheProviders = ({ children }) => {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          {children}
        </QueryClientProvider>
      </Provider>
    </BrowserRouter>
  );
};
const renderWithContext = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { renderWithContext as render };
