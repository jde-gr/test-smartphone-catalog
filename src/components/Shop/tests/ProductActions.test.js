import { render } from '../../../test-utils/testing-library-utils';
import ProductActions from '../ProductActions';

test('renders correctly', () => {
  const { container } = render(<ProductActions />);
  expect(container).toMatchSnapshot();
});
