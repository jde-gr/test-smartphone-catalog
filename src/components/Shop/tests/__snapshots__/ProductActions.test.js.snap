// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="css-1xe7ryq"
  >
    <form
      action="#"
    >
      <div
        id="storageGroup"
      >
        Almacenamiento: 
      </div>
      <div
        aria-labelledby="storageGroup"
        role="group"
      />
      <div
        id="colorGroup"
      >
        Color: 
      </div>
      <div
        aria-labelledby="colorGroup"
        role="group"
      />
      <button
        class="css-1p3dwju"
        type="submit"
      >
        AÃ±adir al carrito
      </button>
    </form>
  </div>
</div>
`;
